cypress/e2e/awx/resources/workflowVisualizer.cy.ts

```diff
@@ -166,54 +141,274 @@ describe('Workflow Job templates visualizer', () => {
-          organization = org;
-          cy.createAwxProject({ organization: organization.id }).then((p) => {
-            project = p;
-            cy.createAwxInventorySource(inventory, project).then((invSrc) => {
-              inventorySource = invSrc;
-              cy.createAwxWorkflowVisualizerInventorySourceNode(
-                workflowJobTemplate,
-                inventorySource
-              ).then((inventorySourceNode) => {
-                cy.createAwxWorkflowVisualizerManagementNode(workflowJobTemplate, 1).then(
-                  (managementNode) => {
-                    cy.createWorkflowJTSuccessNodeLink(projectNode, inventorySourceNode);
-                    cy.createWorkflowJTAlwaysNodeLink(inventorySourceNode, managementNode);
-                  }
-                );
-              });
-            });
-          });
-        });
-      });
+    cy.createAwxWorkflowVisualizerProjectNode(workflowJobTemplate, project).then((projectNode) => {
+      cy.createAwxWorkflowVisualizerInventorySourceNode(workflowJobTemplate, inventorySource).then(
+        (inventorySourceNode) => {
+          cy.createAwxWorkflowVisualizerManagementNode(workflowJobTemplate, 1).then(
+            (managementNode) => {
+              cy.createWorkflowJTSuccessNodeLink(projectNode, inventorySourceNode);
+              cy.createWorkflowJTAlwaysNodeLink(inventorySourceNode, managementNode);
+            }
+          );
+        }
+      );
       cy.visit(`/templates/workflow_job_template/${workflowJobTemplate?.id}/visualizer`);
-      cy.get('[data-cy="wf-vzr-title"]')
-        .should('contain', 'Workflow Visualizer')
-        .should('be.visible');
       cy.get('[data-cy="wf-vzr-name"]')
         .should('contain', `${workflowJobTemplate.name}`)
         .should('be.visible');
-      //remove-node, add-node-and-link, add-link, add-link, edit-node
       cy.removeAllNodesFromVisualizerToolbar();
       cy.contains('button', 'Save').click();
       cy.deleteAwxWorkflowJobTemplate(workflowJobTemplate);
     });
   });
+  it('Can edit a node resource on a workflow visualizer already containing existing nodes', function () {
+    cy.createAwxWorkflowVisualizerProjectNode(workflowJobTemplate, project).then((projectNode) => {
+      cy.createAwxWorkflowVisualizerInventorySourceNode(workflowJobTemplate, inventorySource).then(
+        (inventorySourceNode) => {
+          cy.createAwxWorkflowVisualizerManagementNode(workflowJobTemplate, 1)
+            .then((managementNode) => {
+              cy.createWorkflowJTSuccessNodeLink(projectNode, inventorySourceNode);
+              cy.createWorkflowJTAlwaysNodeLink(inventorySourceNode, managementNode);
+            })
+            .then(() => {
+              cy.visit(`/templates/workflow_job_template/${workflowJobTemplate?.id}/visualizer`);
+              cy.get(`g[data-id=${projectNode.id}] .pf-topology__node__action-icon`).click({
+                force: true,
+              });
```

Comment:
Can you add a baseline assertion here to show that the initial value of the node is type project? that provides a comparison for later when the node is edited.